### PR TITLE
add communicating test with kubernetes apiserver

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: Zou Nengren  zouyee1989@gmail.com

## Description

Currently， we could not check the health of client, it is a reasonable to support the health check.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required]
#294
(https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Do we need this? If so, shouldn't it be part of a long-running go-routine that indicates the status via /healthz?

File: yaml/gateway-dep.yml [hunk 1]
```diff
@@ -59,6 +59,13 @@ spec:
       - name: faas-netes
         image: openfaas/faas-netes:0.6.0
         imagePullPolicy: Always
+        livenessProbe:
```
Comment (alexellis) [inline yaml/gateway-dep.yml]: This would have to be added to the helm chart also.

**Comment by stefanprodan** [comment]
@alexellis why we need this check in the livenessProbe? A controller should survive a short k8s API downtime without crash looping. With this change when you do control plane upgrade faas-netes will crash.

File: yaml/gateway-dep.yml (hunk 1 above)
Comment (zouyee) [reply yaml/gateway-dep.yml]: done

**Comment by alexellis** [comment]
@stefanprodan I'm not sure why you're asking me that question? I suggested that we don't need this PR at all. What do we want to do?

**Pull request closed without merging**